lib/rack/request.rb

```diff
@@ -489,37 +584,24 @@ def parse_multipart
         Rack::Multipart.extract_multipart(self, query_parser)
       end
 
-      def split_ip_addresses(ip_addresses)
-        ip_addresses ? ip_addresses.strip.split(/[,\s]+/) : []
-      end
-
-      def hostname
-        if forwarded = get_header(HTTP_X_FORWARDED_HOST)
-          forwarded.split(/,\s?/).last
-        else
-          get_header(HTTP_HOST) ||
-            get_header(SERVER_NAME) ||
-            get_header(SERVER_ADDR)
-        end
+      def split_header(value)
+        value ? value.strip.split(/[,\s]+/) : []
       end
 
-      def strip_port(ip_address)
-        # IPv6 format with optional port: "[2001:db8:cafe::17]:47011"
-        # returns: "2001:db8:cafe::17"
-        sep_start = ip_address.index('[')
-        sep_end = ip_address.index(']')
-        if (sep_start && sep_end)
-          return ip_address[sep_start + 1, sep_end - 1]
-        end
+      AUTHORITY = /(?<host>(\[(?<ip6>.*)\])|(?<ip4>[\d\.]+)|(?<name>[a-zA-Z0-9\.\-]+))(:(?<port>\d+))?/
+      private_constant :AUTHORITY
```

Comment:
This appears to be only used in one place.  I think it should be inlined into the method, and then you can use it with `=~` so that the named captures become local variables.

```ruby
/(?<host>(\[(?<ip6>.*)\])|(?<ip4>[\d\.]+)|(?<name>[a-zA-Z0-9\.\-]+))(:(?<port>\d+))?/ =~ authority
```

I would also use `ipv6` and `ipv4` as the capture names.

Reply:
If you use it in the method, doesn't the regexp get compiled multiple times?

Reply:
No, regexps without interpolation are static.

Reply:
Okay, I'll try to rework it.

Reply:
It's quite a big regex so I added documentation and made it `/x`.